pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -549,16 +549,31 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 		Name:     nicName,
 		VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
 	}
-	switch set := s.scope.Machine.ObjectMeta.Labels[v1alpha1.MachineRoleLabel]; set {
-	case v1alpha1.Node:
-		networkInterfaceSpec.SubnetName = azure.GenerateNodeSubnetName(s.scope.Cluster.Name)
-	case v1alpha1.ControlPlane:
-		networkInterfaceSpec.SubnetName = azure.GenerateControlPlaneSubnetName(s.scope.Cluster.Name)
-		networkInterfaceSpec.PublicLoadBalancerName = azure.GeneratePublicLBName(s.scope.Cluster.Name)
-		networkInterfaceSpec.InternalLoadBalancerName = azure.GenerateInternalLBName(s.scope.Cluster.Name)
-		networkInterfaceSpec.NatRule = 0
-	default:
-		return errors.Errorf("unknown value %s for label `set` on machine %s, skipping machine creation", set, s.scope.Machine.Name)
+
+	// If any of the following fields is set, ignore role based setting
+	if s.scope.MachineConfig.Subnet != "" || s.scope.MachineConfig.PublicLoadBalancer != "" || s.scope.MachineConfig.InternalLoadBalancer != "" {
+		if s.scope.MachineConfig.Subnet != "" {
+			networkInterfaceSpec.SubnetName = s.scope.MachineConfig.Subnet
+		}
+		if s.scope.MachineConfig.PublicLoadBalancer != "" {
+			networkInterfaceSpec.PublicLoadBalancerName = s.scope.MachineConfig.PublicLoadBalancer
+			networkInterfaceSpec.NatRule = 0
+		}
+		if s.scope.MachineConfig.InternalLoadBalancer != "" {
+			networkInterfaceSpec.InternalLoadBalancerName = s.scope.MachineConfig.InternalLoadBalancer
+		}
+	} else {
+		switch set := s.scope.Machine.ObjectMeta.Labels[v1alpha1.MachineRoleLabel]; set {
+		case v1alpha1.Node:
+			networkInterfaceSpec.SubnetName = azure.GenerateNodeSubnetName(s.scope.Cluster.Name)
+		case v1alpha1.ControlPlane:
+			networkInterfaceSpec.SubnetName = azure.GenerateControlPlaneSubnetName(s.scope.Cluster.Name)
+			networkInterfaceSpec.PublicLoadBalancerName = azure.GeneratePublicLBName(s.scope.Cluster.Name)
+			networkInterfaceSpec.InternalLoadBalancerName = azure.GenerateInternalLBName(s.scope.Cluster.Name)
+			networkInterfaceSpec.NatRule = 0
```

Comment:
ah! been wanting to get rid of this Natrules as well, since ssh for masters are not enabled by default on openshift. Probably better to provide the natrule as an input